scripts/build_shapes.py

```diff
@@ -80,7 +81,57 @@ def download_GADM(country_code, update=False, out_logging=False):
     return GADM_inputfile_gpkg, GADM_filename
 
 
-def get_GADM_layer(country_list, layer_id, geo_crs, update=False, outlogging=False):
+def replace_nonstd_codes(row, col, country_code, keep_cond=True):
+    if row["GID_0"] != country_code:
+        return country_name_2_two_digits(row["COUNTRY"])
+    else:
+        return row["GID_0"]
+
+
+def filter_gadm(geodf, layer, cc, output_nonstd_to_csv=True, keep_all_areas=True):
+
+    # convert country name representation of the main country (GID_0 column)
+    geodf["GID_0"] = geodf["GID_0"].map(three_2_two_digits_country)
```

Comment:
I'd remove this line and having the filtering being done by time below in the ifs; it may be redundant.

However, we need to move the block to save the nonstandard codes here otherwise the ifs overwrite the data.
Please, revise the check of the if with the appropriate parameter (output_nonstd_to_csv), whose default value shall be false.

Probably, non standard ids may be got similarly to below as:
`geodf[geodf["GID_0"] != two_2_three_digits_country(cc)].index`

Reply:
Not quite sure I understand :)
That is not filtering but ISO2 -> ISO3 transformation

Reply:
My feeling is that this first line is not necessary.
The scope of that line is to preprocess the column to identify what areas are contended zones by checking those that are converted into "not found".

However, since(please confirm) the content of GID_0 is ISO3 code, then the code `geodf["GID_0"] != two_2_three_digits_country(cc)` should do the same without performing that preprocessing.

Applying the `three_2_two_digits_country` may only slow down the computational time.

Reply:
Thanks for the explanations!

If the workflow will work properly using directly GID_0 as ISO3 code instead transforming it into ISO2, then agree that the first line is not needed. I was following the original code when keeping this transformation:

https://github.com/pypsa-meets-earth/pypsa-earth/blob/7f6eb6ad0af1fe31dd37d7cee6a642e190bd1893/scripts/build_shapes.py#L117-L120

My understanding was that #262 issue means that we need ISO2 code everywhere but I can be wrong